src/algorithms/dominators.rs

```diff
@@ -49,13 +103,27 @@ pub struct DominatorTree {
 }
 
 impl DominatorTree {
-    fn new(graph: &PortGraph, entry: NodeIndex, direction: Direction) -> Self {
+    fn new(
+        graph: &PortGraph,
+        entry: NodeIndex,
+        direction: Direction,
+        mut node_filter: impl FnMut(NodeIndex) -> bool,
+        mut port_filter: impl FnMut(NodeIndex, PortIndex) -> bool,
```

Comment:
Here we don't store the closures, so the monomorphization can inline trivial functions (like the non-filtered call) and optimize the checks away.

In contrast, the iterator structures from the others need to store the closures as fat pointers, so without the `Options` it would always have to do dynamic dispatch even for trivial filters.